File: Application/src/main/java/Repository/ArtistRepo.java
package Repository;

import Model.Artist;
import java.sql.*;
import java.util.ArrayList;
import java.util.List;

public class ArtistRepo
{
    private static final String DB_URL = "jdbc:sqlite:database.sqlite";


    public void addArtist(Artist artist) throws SQLException
    {
        String sql = "INSERT INTO Artist (name, birth_date, birth_place, nationality) VALUES (?, ?, ?, ?)";

        try (Connection conn = DriverManager.getConnection(DB_URL);
             PreparedStatement stmt = conn.prepareStatement(sql)) {
            stmt.setString(1, artist.getName());
            stmt.setString(2, artist.getBirthDate());
            stmt.setString(3, artist.getBirthPlace());
            stmt.setString(4, artist.getNationality());
            stmt.executeUpdate();
        }
    }


    public List<Artist> getAllArtists() throws SQLException
    {
        List<Artist> artists = new ArrayList<>();
        String sql = "SELECT name, birth_date, birth_place, nationality FROM Artist";

        try (Connection conn = DriverManager.getConnection(DB_URL);
             Statement stmt = conn.createStatement();
             ResultSet rs = stmt.executeQuery(sql)) {
            while (rs.next()) {
                artists.add(new Artist(
                        rs.getString("name"),
                        rs.getString("birth_date"),
                        rs.getString("birth_place"),
                        rs.getString("nationality")
                ));
            }
        }
        return artists;
    }


    public void deleteArtist(String name) throws SQLException
    {
        String sql = "DELETE FROM Artist WHERE name = ?";

        try (Connection conn = DriverManager.getConnection(DB_URL);
             PreparedStatement stmt = conn.prepareStatement(sql)) {
            stmt.setString(1, name);
            stmt.executeUpdate();
        }
    }


    public void updateArtist(String oldName, Artist updatedArtist) throws SQLException
    {
        String sql = "UPDATE Artist SET name = ?, birth_date = ?, birth_place = ?, nationality = ? WHERE name = ?";

        try (Connection conn = DriverManager.getConnection(DB_URL);
             PreparedStatement stmt = conn.prepareStatement(sql)) {
            stmt.setString(1, updatedArtist.getName());
            stmt.setString(2, updatedArtist.getBirthDate());
            stmt.setString(3, updatedArtist.getBirthPlace());
            stmt.setString(4, updatedArtist.getNationality());
            stmt.setString(5, oldName);
            stmt.executeUpdate();
        }
    }


    public Artist findArtistByName(String name) throws SQLException
    {
        String sql = "SELECT * FROM Artist WHERE name = ?";

        try (Connection conn = DriverManager.getConnection(DB_URL);
             PreparedStatement stmt = conn.prepareStatement(sql)) {
            stmt.setString(1, name);
            ResultSet rs = stmt.executeQuery();
            if (rs.next()) {
                return new Artist(rs.getString("name"), rs.getString("birth_date"),
                        rs.getString("birth_place"), rs.getString("nationality"));
            }
            return null;
        }
    }
}

File: AssignmentMVVM/Model/Repository/ArtistRepo.cs
using System.Data.SqlClient; 
using AssignmentMVVM.Model;

namespace AssignmentMVVM.Repository
{
    public class ArtistRepo
    {
        private static readonly string DB_URL = "Data Source=(LocalDB)\\MSSQLLocalDB;Initial Catalog=art_gallery;Integrated Security=True;";


        public void AddArtist(Artist artist)
        {
            string sql = "INSERT INTO Artist (name, birth_date, birth_place, nationality, photo) VALUES (@name, @birthDate, @birthPlace, @nationality, @photo)";
            using (var conn = new SqlConnection(DB_URL))
            {
                conn.Open();
                using (var cmd = new SqlCommand(sql, conn))
                {
                    cmd.Parameters.AddWithValue("@name", artist.Name);
                    cmd.Parameters.AddWithValue("@birthDate", artist.BirthDate);
                    cmd.Parameters.AddWithValue("@birthPlace", artist.BirthPlace);
                    cmd.Parameters.AddWithValue("@nationality", artist.Nationality);
                    cmd.Parameters.AddWithValue("@photo", artist.Photo ?? (object)DBNull.Value);
                    cmd.ExecuteNonQuery();
                }
            }
        }


        public List<Artist> GetAllArtists()
        {
            List<Artist> artists = new List<Artist>();
            string sql = "SELECT name, birth_date, birth_place, nationality, photo FROM Artist";

            using (var conn = new SqlConnection(DB_URL))
            {
                conn.Open();
                using (var cmd = new SqlCommand(sql, conn))
                using (var rs = cmd.ExecuteReader())
                {
                    while (rs.Read())
                    {
                        var photoValue = rs.IsDBNull(4) ? null : rs.GetString(4);
                        // debug
                        Console.WriteLine($"Loaded artist photo path: {photoValue}");

                        artists.Add(new Artist(
                            rs.GetString(0),
                            rs.GetDateTime(1),
                            rs.GetString(2),
                            rs.GetString(3),
                            photoValue
                        ));
                    }
                }
            }

            return artists;
        }


        public void DeleteArtist(string name)
        {
            string sql = "DELETE FROM Artist WHERE name = @name";
            using (var conn = new SqlConnection(DB_URL)) 
            {
                conn.Open();
                using (var cmd = new SqlCommand(sql, conn)) 
                {
                    cmd.Parameters.AddWithValue("@name", name);
                    cmd.ExecuteNonQuery();
                }
            }
        }

        public void UpdateArtist(string oldName, Artist updatedArtist)
        {
            string sql = "UPDATE Artist SET name = @name, birth_date = @birthDate, birth_place = @birthPlace, nationality = @nationality, photo = @photo WHERE name = @oldName";
            using (var conn = new SqlConnection(DB_URL)) 
            {
                conn.Open();
                using (var cmd = new SqlCommand(sql, conn)) 
                {
                    cmd.Parameters.AddWithValue("@name", updatedArtist.Name);
                    cmd.Parameters.AddWithValue("@birthDate", updatedArtist.BirthDate);
                    cmd.Parameters.AddWithValue("@birthPlace", updatedArtist.BirthPlace);
                    cmd.Parameters.AddWithValue("@nationality", updatedArtist.Nationality);
                    cmd.Parameters.AddWithValue("@photo", updatedArtist.Photo ?? (object)DBNull.Value);
                    cmd.Parameters.AddWithValue("@oldName", oldName);
                    cmd.ExecuteNonQuery();
                }
            }
        }

        public Artist FindArtistByName(string name)
        {
            string sql = "SELECT name, birth_date, birth_place, nationality, photo FROM Artist WHERE name = @name";
            using (var conn = new SqlConnection(DB_URL)) 
            {
                conn.Open();
                using (var cmd = new SqlCommand(sql, conn)) 
                {
                    cmd.Parameters.AddWithValue("@name", name);
                    using (var rs = cmd.ExecuteReader())
                    {
                        if (rs.Read())
                        {
                            return new Artist(
                                rs.GetString(0),
                                rs.GetDateTime(1), 
                                rs.GetString(2),
                                rs.GetString(3),
                                rs.GetString(4)
                            );
                        }
                    }
                }
            }
            return null;
        }

        public List<Artist> SearchByName(string name)
        {
            List<Artist> artists = new List<Artist>();
            string sql = "SELECT name, birth_date, birth_place, nationality, photo FROM Artist WHERE name LIKE @name"; 
            using (var conn = new SqlConnection(DB_URL)) 
            {
                conn.Open();
                using (var cmd = new SqlCommand(sql, conn)) 
                {
                    cmd.Parameters.AddWithValue("@name", $"%{name}%");
                    using (var rs = cmd.ExecuteReader())
                    {
                        while (rs.Read())
                        {
                            artists.Add(new Artist(
                                rs.GetString(0),
                                rs.GetDateTime(1), 
                                rs.GetString(2),
                                rs.GetString(3),
                                rs.GetString(4)
                            ));
                        }
                    }
                }
            }
            return artists;
        }
    }
}